src/views/dashboard/modules/data-view/components/data-grid.vue

```diff
@@ -28,25 +50,77 @@ a-card(:bordered="false")
     'show-page-size': true,
   }
 
-  const gridColumn = computed(() => {
+  // replace '.' with '-' to make it a valid data-index
+  // useful when '.' is used in column name, such as `SELECT 1.1`, `SELECT 1 AS a.b`
+  function columnNameToDataIndex(columnName: string) {
+    return columnName.replace(/\./gi, '-')
+  }
+
+  const timeColumnNames = computed(() => {
+    return props.data.records.schema.column_schemas
```

Comment:
```suggestion
    const { schema } = props.data.records
    if (!schema) return []
    return props.data.records.schema.column_schemas
```

Reply:
I have just realized this would cause a problem if the results are empty. (This is similar to `gridColumns`)

Reply:
Typescript would give a hint if we have typed it correctly before 😇

Reply:
> Typescript would give a hint if we have typed it correctly before 😇

What hint?

Reply:
If `props.data` is typed correctly, then Typescript would thorw an error that `props.data.records.schema` is optional and we have to check if it is exists

Reply:
> If `props.data` is typed correctly, then Typescript would throw an error that `props.data.records.schema` is optional and we have to check if it is exists

I see...You are absolutely right. This is a problem caused by old bad habits and we should have used Typescript better. I would do more refactoring in the future.